### ISSUE TITLE
MA0075 false positive: `FormattableString.Invariant($"...")` used as operand of string `+`

### ISSUE THREAD

**Issue by Bouke** [meziantou/Meziantou.Analyzer#1321]
## Version of the Meziantou.Analyzer NuGet package

3.0.177 (also reproduces on 3.0.153; 3.0.117 is not affected)

## Rule Identifier

MA0075

## Target Framework

net10.0

## C# Language version

C# 14.0 (compiler 5.0.0-2.25523.111)

## .NET SDK version

10.0.100

## Description

Since 3.0.150 (PR #1273, "Improve culture-sensitive formatting analysis for polymorphic values"), MA0075 reports `FormattableString.Invariant($"{0:N0}")` when the result is used directly as an operand of string concatenation. `Invariant` returns a `string` formatted with the invariant culture, so there is no implicit culture-sensitive `ToString`.

Storing the result in a local first (`var a = Invariant(...); _ = a + b;`) is already accepted, so the inline form should be too. 3.0.117 does not report this; 3.0.153 and 3.0.177 do.

## Reproduction Steps

Minimal code (top-level statements, `net10.0`, `dotnet_diagnostic.MA0075.severity = warning` in `.editorconfig`):

```c#
using static System.FormattableString;

_ = Invariant($"{0:N0}") + Invariant($"{1:N0}"); // MA0075 on both operands
_ = Invariant($"{0:N0}") + "x";                  // MA0075 on left operand
_ = "x" + Invariant($"{1:N0}");                  // MA0075 on right operand
_ = (Invariant($"{0:N0}")) + "x";                // MA0075

// Not reported (expected):
var a = Invariant($"{0:N0}");
var b = Invariant($"{1:N0}");
_ = a + b;
_ = string.Concat(Invariant($"{0:N0}"), "x");
_ = Invariant($"{0:N0}").ToUpperInvariant() + "x";
```

Output with 3.0.177:

```
Program.cs(3,5):  warning MA0075: Do not use implicit culture-sensitive ToString
Program.cs(3,28): warning MA0075: Do not use implicit culture-sensitive ToString
Program.cs(4,5):  warning MA0075: Do not use implicit culture-sensitive ToString
Program.cs(5,11): warning MA0075: Do not use implicit culture-sensitive ToString
Program.cs(6,6):  warning MA0075: Do not use implicit culture-sensitive ToString
```

The same code with 3.0.117 reports nothing.

## Other information

**Cause.** `CultureSensitiveFormattingContext.GetCultureSensitivity(IOperation, ...)` for an `IInvocationOperation` calls `GetInterpolatedStringArgumentCultureSensitivity(invocation, options)`. That helper treats *any* invocation with a `FormattableString` (or interpolated-string-handler) argument as producing the culture sensitivity of that argument, and returns before the `UseInvocationReturnType` check:

```c#
// Check interpolated string arguments that are formatted by the invocation.
var interpolatedStringArgumentCultureSensitivity = GetInterpolatedStringArgumentCultureSensitivity(invocation, options);
if (interpolatedStringArgumentCultureSensitivity is not null)
    return interpolatedStringArgumentCultureSensitivity.Value;

if ((options & CultureSensitiveOptions.UseInvocationReturnType) == CultureSensitiveOptions.UseInvocationReturnType)
    return GetCultureSensitivity(invocation.Type, options);
```

For `Invariant($"{0:N0}")` the argument `$"{0:N0}"` is culture-sensitive (an `int` with format `N0`), so the whole invocation is classified as culture-sensitive, even though `Invariant` exists precisely to format that argument with `CultureInfo.InvariantCulture`. `Invariant` is not in `_excludedMethods` and has no `IFormatProvider` argument, so nothing short-circuits before this point.

MA0076 already handles this case correctly by checking for a conversion to `FormattableString` in `AnalyzeInterpolatedString`, which is why the interpolated string itself is not reported; only the enclosing `+` analysis (MA0075) regresses.

**Not a duplicate of #1316.** That issue was about a custom wrapper over `DefaultInterpolatedStringHandler`, which formats eagerly at the call site with the current culture, so the diagnostic there was correct. `FormattableString` is different: it defers formatting and captures the arguments, and `FormattableString.Invariant` formats them with `CultureInfo.InvariantCulture`. The result is genuinely culture-insensitive. The `[CultureInsensitive]` attribute added in #1318 does not apply here, since `Invariant` is a BCL method. The code path above is unchanged at commit 22602be (the #1318 merge), so the issue is still present on `main`.

**Suggested fix.** Treat `System.FormattableString.Invariant(FormattableString)` (and `FormattableString.ToString(IFormatProvider)`, `CurrentCulture`, `string.Create(IFormatProvider, ...)`) as culture-insensitive before consulting the interpolated-string argument, e.g. add `Invariant` to `CreateExcludedMethods`, or have `GetInterpolatedStringArgumentCultureSensitivity` skip methods whose containing type is `System.FormattableString`.

A test similar to the existing one at `DoNotUseImplicitCultureSensitiveToStringAnalyzerTests.cs:302` (`string a = System.FormattableString.Invariant($"abc{1}");`) but with the result used in a concatenation would cover it:

```c#
string a = System.FormattableString.Invariant($"abc{1:N0}") + "x";
```

Analyzer source inspected at commit 22602be (2026-08-20). Environment: macOS 15 arm64.

---

<sub>This report was prepared with the assistance of Claude Fable 5 (Anthropic), which reproduced the issue, bisected the regression, and traced the cause in the analyzer source. I reviewed and verified the findings.</sub>
